pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -202,19 +205,48 @@ func GatherClusterOperators(i *Gatherer) func() ([]record.Record, []error) {
 				continue
 			}
 			for _, namespace := range namespacesForOperator(&item) {
+				if namespaceEventsCollected.Has(namespace) {
+					continue
+				}
 				pods, err := i.coreClient.Pods(namespace).List(metav1.ListOptions{})
 				if err != nil {
 					klog.V(2).Infof("Unable to find pods in namespace %s for failing operator %s", namespace, item.Name)
 					continue
 				}
-				for i := range pods.Items {
-					if isHealthyPod(&pods.Items[i], now) {
+				for j := range pods.Items {
+					pod := &pods.Items[j]
+					if isHealthyPod(pod, now) {
 						continue
 					}
-					records = append(records, record.Record{Name: fmt.Sprintf("config/pod/%s/%s", pods.Items[i].Namespace, pods.Items[i].Name), Item: PodAnonymizer{&pods.Items[i]}})
-				}
-				if namespaceEventsCollected.Has(namespace) {
-					continue
+					records = append(records, record.Record{Name: fmt.Sprintf("config/pod/%s/%s", pod.Namespace, pod.Name), Item: PodAnonymizer{pod}})
+
+					// Fetch previous and current pod logs
+					for _, previous := range []bool{true, false} {
+						for _, container := range pod.Spec.Containers {
```

Comment:
I was looking at this in 4.5 and master and there we use `pod.Spec.InitContainers`

Reply:
Good idea, backported release-4.5 parts here as well

Reply:
Thanks for quick update. I'll check ASAP.